tldw_Server_API/app/api/v1/endpoints/mcp_unified_endpoint.py

```
@@ -1698,6 +1699,36 @@ async def _probe_mcp_connection(url: str, headers: dict[str, str]) -> None:
     resp.raise_for_status()
 
 
+def _validate_public_mcp_url(raw_url: str) -> str:
+    """Return a normalized public HTTP(S) URL or raise ValueError."""
+    raw = str(raw_url or "").strip()
+    parsed = urlparse(raw)
+    if parsed.scheme not in ("http", "https"):
+        raise ValueError("Only http and https URLs are allowed")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)